correct_periscope_drift.py

```diff
@@ -0,0 +1,431 @@
+    return mp
+
+
+# The '@handle_ciao_errors' decorator will catch any error thrown and
+# display it in a format like that of a CIAO tool, then exit with a
+# non-zero status. So, if at any point in your code you need to exit just
+# raise an error and let handle_ciao_errors bother with the display.
+#
+# This also means that routines can be used from ChIPS/Sherpa/other scripts
+# without having to worry about them calling sys.exit.
+
+
+@handle_ciao_errors(TOOLNAME, VERSION)
+def main(opt):
+    events = extract_events(opt['evtfile'],
+                            opt['x'], opt['y'], opt['radius'])
+
+    evt_ra_nom = events.get_key('RA_NOM').value
+    evt_dec_nom = events.get_key('DEC_NOM').value
+    evt_roll_nom = events.get_key('ROLL_NOM').value
+
+    asol = pycrates.read_file(opt['infile'])
+    asol_times = asol.get_column('time').values
+
+    # Sanity check the two input files
+    asol_obsid = asol.get_key('OBS_ID').value
+    evt_obsid = events.get_key('OBS_ID').value
+    if asol_obsid != evt_obsid:
+        v1("Error Aspect solution obsid {} != event file obsid {}".format(asol_obsid, evt_obsid))
+
+    evt_ra = events.get_column('RA').values
+    evt_dec = events.get_column('Dec').values
+    evt_times = events.get_column('Time').values
+
+    # Limit to only using events contained within the range of the aspect solution
+    ok_times = (evt_times > asol_times[0]) & (evt_times < asol_times[-1])
+    if not np.any(ok_times):
+        raise ValueError("No events in region are contained within time range of aspect solution.")
+    # Limit this *in place*
+    evt_ra = evt_ra[ok_times]
+    evt_dec = evt_dec[ok_times]
+    evt_times = evt_times[ok_times]
+
+    if len(evt_times) < opt['src_min_counts']:
+        v1("Warning only {} counts in src region.  {} minimum suggested 'src_min_counts'".format(
+                len(evt_times), opt['src_min_counts']))
+
+    ax_data = {}
+    ax_map = {'yag': 'dy',
+              'zag': 'dz'}
+
+    ax_data['yag'], ax_data['zag'] = get_event_yag_zag(evt_ra, evt_dec,
+                                                       evt_ra_nom, evt_dec_nom, evt_roll_nom)
+
+    # Store comments to print in block after all of the sherpa fit output
+    fit_comments = []
+    plot_list = []
+
+    for data_id, ax in enumerate(['yag', 'zag']):
+        fit_data = ax_data[ax] - np.mean(ax_data[ax])
+        mp = fit(fit_data, evt_times, data_id, opt)
+
+        bin_centers, bin_mean, bin_std = time_bins(evt_times, fit_data)
+
+        add_window()
+        add_curve((bin_centers - evt_times[0]) / 1000., bin_mean, [bin_std, +bin_std],
+                  ["line.style", "none", "symbol.style", "none", "err.style", "cap"])
+        add_curve(mp.x / 1000., mp.y, ["symbol.style", "none"])
+        set_plot_xlabel("Observation elapsed/delta time (ks)")
+        set_plot_ylabel("Position offset from mean, {} (arcsec)".format(ax))
+        set_plot_title("Fit of {} data (with time-binned event offsets)".format(ax))
```

Comment:
"yag" and "zag" labels should be replaced with something like "Aspect Camera Y axis" and this should be more clear in the docs.